### PR TITLE
[JENKINS-56868] Fix getting information on AIX processes not owned by the current user

### PR DESCRIPTION
**Pull request by robert-shade**
See [JENKINS-56868](https://issues.jenkins-ci.org/browse/JENKINS-56868).

### Proposed changelog entries

* Fix getting information on AIX processes not owned by the current user

### CODE REVIEW

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 1]
```diff
@@ -1155,25 +1095,88 @@ private AIXProcess(int pid) throws IOException {
...
+                            return true;
+                        } else {
+                            throw new IOException("Unrecognized data model value"); // sanity check
+                        }
+                    } finally {
+                        pstatus.close();
+                    }
+                } catch (IOException | LastErrorException e) {
+                    // failed to read. this can happen under normal circumstances (most notably permission denied)
+                    // so don't report this as an error.
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/ProcessTree.java]: Would be nice to have some system logging at least, e.g. on the FINE level

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 2]
```diff
@@ -1155,25 +1095,88 @@ private AIXProcess(int pid) throws IOException {
...
                     pr_argp = adjustL(psinfo.readLong());
                     pr_envp = adjustL(psinfo.readLong());
+
+                    b64 = is64BitProcess();
                 } finally {
                     psinfo.close();
                 }
             }
 
+            protected boolean is64BitProcess() {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/ProcessTree.java]: Do you plan to have any API consumers outside? If not, it might be better to just make it private

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Minor changes are needed before landing this PR.
We do not have any kind of test automation for AIX, but such changes are OK since they do not put other platforms at risk

**Comment by oleg-nenashev** [comment]
@robert-shade Are you still interested in it?

**Comment by MRamonLeon** [comment]
@robert-shade could you address @oleg-nenashev's request? In another way, we need to close this PR. Thanks!

**Comment by fcojfernandez** [comment]
Since there hasn't been more activity on this issue I'm closing the PR. Please feel free to re-open if you can work on this again and address the feedback from Oleg.

**Pull request closed without merging**